### ISSUE TITLE
Repeated timeline, many read receipts

### ISSUE THREAD

**Issue by timokoesters** [vector-im/element-android#5166]
### Steps to reproduce

Log in to a conduit account.
Accept a federated invite into a new room.

The problem seems to be related to the limited flag for syncs or the /messages endpoint.

### Outcome

<img src="https://user-images.githubusercontent.com/25297359/152646539-80277510-e05b-41b0-8456-3715739115a9.png" width="540">

### Your phone model

-

### Operating system version

-

### Application version and app store

-

### Homeserver

conduit.rs

### Will you send logs?

No

**Comment by BillCarsonFr** [comment]
So back pagination is like reloading the same page again
It's only when accepting a federated invite into a new room?

**Comment by timokoesters** [comment]
It's easy to reproduce when you are at the "start" of a timeline (/messages doesn't return any new events), but it seems to happen in other cases too.

**Comment by BillCarsonFr** [comment]
There is a difference between an empty pagination result on conduit.rs and synapse.

Synapse
```
{"chunk":[],"start":"t1-1339_71365_114_338_608_1_436_640_1","end":"t1-1339_71365_114_338_608_1_436_640_1"}
```

Conduit
```
{"start":"19057171","chunk":[]}
```

Looks like somehow android is expected the end to be set, wich as per spec shouldn't

`end`
> If no further events are available (either because we have reached the start of the timeline, or because the user does not have permission to see any more events), this property is omitted from the response.

**Comment by BillCarsonFr** [comment]
Quikly confirmed by updating android code to use start for end when chunk is empty and it's working fine

**Comment by HarHarLinks** [comment]
I've hit something very similar on SchildiChat based on 1.4.4 with synapse 1.54.0.
I opened a room, clicked to jump to first unread, tried to scroll down from there but the timeline would repeat itself endlessly from there.
<details><summary>Expand to show a screenshot of the loop</summary>

![image](https://user-images.githubusercontent.com/2803622/159129113-6e910fe0-bf01-487f-ae82-366b9b09aeb4.png)

</details>
I did rageshake this (to the schildi devs).

**Comment by MadLittleMods** [comment]
I reproduced a repeating never-ending timeline a lot when I had to use Element Android on a trip. Here is a rageshake, https://github.com/matrix-org/element-android-rageshakes/issues/35689. This was with normal messages, not just state events and read receipts.

**Comment by MrAureliusR** [comment]
This bit me when self-hosting a Conduit server. It's just an endless scroll of the same messages over and over, and duplicated read receipts.
![element_issue](https://user-images.githubusercontent.com/41264289/166481976-7b86b9ad-a369-4634-a13a-c36f50ecb4de.png)
.

**Comment by SpiritCroc** [comment]
I wonder if this issue will be fixed with https://github.com/vector-im/element-android/pull/5878/commits/638e18eaab475744b3a23742bceb9c2f2319e66c .

That PR and some more experimental changes for timeline loops are already included in the latest SchildiChat beta (1.4.13.sc53-test1), if anybody is interested in testing it.

**Comment by MrAureliusR** [comment]
> I wonder if this issue will be fixed with [638e18e](https://github.com/vector-im/element-android/commit/638e18eaab475744b3a23742bceb9c2f2319e66c) from #5878 .
> 
> That PR and some more experimental changes for timeline loops are already included in the latest SchildiChat beta (1.4.13.sc53-test1), if anybody is interested in testing it.

Just downloaded SchildiChat on my Android device and joined the beta. After it updated to the beta version, signed into my homeserver -- exact same issue as with Element.

**Comment by SpiritCroc** [comment]
> > I wonder if this issue will be fixed with [638e18e](https://github.com/vector-im/element-android/commit/638e18eaab475744b3a23742bceb9c2f2319e66c) from #5878 .
> > That PR and some more experimental changes for timeline loops are already included in the latest SchildiChat beta (1.4.13.sc53-test1), if anybody is interested in testing it.
> 
> Just downloaded SchildiChat on my Android device and joined the beta. After it updated to the beta version, signed into my homeserver -- exact same issue as with Element.

Sure you have the right version? (Note we distribute betas on our own fdroid server, google play betas are usually outdated)

**Comment by ganfra** [comment]
This also happens on synapse if you have a version > 1.6 (https://github.com/matrix-org/synapse/pull/12903)
Related: https://github.com/vector-im/element-android/issues/6312
